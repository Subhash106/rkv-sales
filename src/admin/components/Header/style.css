.admin-header {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  list-style: none;
  gap: 1rem;
}

.admin-nav {
  padding: 1rem;
}
