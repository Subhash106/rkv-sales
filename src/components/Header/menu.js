import { func, string } from 'prop-types';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { NavLink } from 'react-router-dom';

export default function Menu({ closeClickHandler, logoutHandler, token }) {
  const { t } = useTranslation();

  return (
    <>
      <li>
        <NavLink
          onClick={closeClickHandler}
          to="/dashboard"
          className={({ isActive }) => {
            return isActive ? 'active' : '';
          }}
          end
        >
          {t('header.dashboard')}
        </NavLink>
      </li>
      <li>
        <NavLink
          onClick={closeClickHandler}
          to="/sales"
          className={({ isActive }) => {
            return isActive ? 'active' : '';
          }}
        >
          {t('header.sales')}
        </NavLink>
      </li>
      <li>
        <NavLink
          onClick={closeClickHandler}
          to="/sales-summary"
          className={({ isActive }) => {
            return isActive ? 'active' : '';
          }}
        >
          {t('header.salesSummary')}
        </NavLink>
      </li>
      <li>
        <NavLink
          onClick={closeClickHandler}
          to="/purchases"
          className={({ isActive }) => {
            return isActive ? 'active' : '';
          }}
        >
          {t('header.purchases')}
        </NavLink>
      </li>
      <li>
        <NavLink
          onClick={closeClickHandler}
          to="/purchases-summary"
          className={({ isActive }) => {
            return isActive ? 'active' : '';
          }}
        >
          {t('header.purchasesSummary')}
        </NavLink>
      </li>
      <li>
        <NavLink
          onClick={closeClickHandler}
          to="/admin"
          className={({ isActive }) => {
            return isActive ? 'active' : '';
          }}
        >
          {t('header.inventory')}
        </NavLink>
      </li>
      {token && (
        <li>
          <a href="#" onClick={logoutHandler}>
            {t('header.logout')}
          </a>
        </li>
      )}
    </>
  );
}

Menu.propTypes = {
  logoutHandler: func,
  closeClickHandler: func,
  token: string
};
