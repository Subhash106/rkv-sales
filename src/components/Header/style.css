.header {
  grid-column: 1/-1;
  border-bottom: 1px solid var(--secondary-gray);

  & .main-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 1.6rem 0;

    & .logo {
      width: 3.5rem;
    }

    & .top-nav {
      & ul {
        list-style: none;
        display: flex;
        justify-content: space-between;
        gap: 3.2rem;
      }
    }
  }
}

.top-nav a:link,
.top-nav a:visited,
.top-nav a.inactive {
  color: var(--primary-gray);
  text-decoration: none;
  font-size: 1.8rem;
  font-weight: 700; 
  display: block;
}

.top-nav a:hover,
.top-nav a:active,
.top-nav a.active {
  color: var(--primary-gray-dark);
}
