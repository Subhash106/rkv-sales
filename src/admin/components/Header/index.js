import React from 'react';
import { useTranslation } from 'react-i18next';
import { NavLink } from 'react-router-dom';

import './style.css';

export default function Header() {
  const { t } = useTranslation();
  return (
    <div className="admin-panel-header">
      <div className="logo-box">
        <NavLink to="/">
          <img src="../../../img/logo.jpg" alt="Logo" className="logo" />
        </NavLink>
      </div>
      <nav className="top-nav">
        <ul className="admin-header">
          <li>
            <NavLink to="/dashboard">{t('header.saleDashboard')}</NavLink>
          </li>
          <li>
            <NavLink to="/admin">{t('header.adminDashboard')}</NavLink>
          </li>
          <li>
            <NavLink to="/items">{t('header.item')}</NavLink>
          </li>
          <li>
            <NavLink to="/colors">{t('header.color')}</NavLink>
          </li>
          <li>
            <NavLink to="/add-inventory">{t('header.addInventory')}</NavLink>
          </li>
          <li>
            <NavLink to="/inventory-summary">{t('header.inventorySummary')}</NavLink>
          </li>
        </ul>
      </nav>
    </div>
  );
}
