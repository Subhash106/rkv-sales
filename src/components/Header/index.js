import React from 'react';
import { NavLink } from 'react-router-dom';
import './style.css';

const Header = () => {
  return (
    <div className="header">
      <div className="container">
        <header className="main-header">
          <div className="logo-box">
            <NavLink to="/">
              <img src="../../../img/logo.jpg" alt="Logo" className="logo" />
            </NavLink>
          </div>

          <div className="top-nav">
            <ul>
              <li>
                <NavLink
                  to="/"
                  className={({ isActive }) => {
                    return isActive ? 'active' : '';
                  }}
                  end
                >
                  Home
                </NavLink>
              </li>
              <li>
                <NavLink
                  to="/orders"
                  className={({ isActive }) => {
                    return isActive ? 'active' : '';
                  }}
                >
                  Orders
                </NavLink>
              </li>
            </ul>
          </div>
        </header>
      </div>
    </div>
  );
};

export default Header;
