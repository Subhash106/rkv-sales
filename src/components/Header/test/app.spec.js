import React from 'react';
import Header from '..';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';

describe('Header test suite', () => {
  it('Should match snapshot', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it('It should have Home navigation', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(NavLink).at(1).text()).toEqual('Home');
  });

  it('It should have Orders navigation', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(NavLink).at(2).text()).toEqual('Orders');
  });
});
