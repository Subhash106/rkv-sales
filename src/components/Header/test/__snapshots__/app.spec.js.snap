// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test suite Should match snapshot 1`] = `ShallowWrapper {}`;
